ivy/functional/backends/jax/data_type.py

```diff
@@ -213,13 +222,15 @@ def as_native_dtype(dtype_in: Union[jnp.dtype, str, bool, int, float], /) -> jnp
         )
 
 
-def dtype(x: JaxArray, *, as_native: bool = False) -> ivy.Dtype:
+def dtype(x: Union[JaxArray, np.ndarray], *, as_native: bool = False) -> ivy.Dtype:
     if as_native:
-        return ivy.to_native(x).dtype
+        return ivy.as_native_dtype(x.dtype)
     return as_ivy_dtype(x.dtype)
 
 
-def dtype_bits(dtype_in: Union[jnp.dtype, str], /) -> int:
+def dtype_bits(dtype_in: Union[jnp.dtype, str, np.dtype], /) -> int:
+    if isinstance(dtype_in, np.dtype):
+        dtype_in = dtype_in.name
     dtype_str = as_ivy_dtype(dtype_in)
```

Comment:
You already modified `as_ivy_dtype` to accept np.dtype why make the conversion here?

Reply:
Yeah makes sense this is redundant